apps/nuxt3-ssr/components/harmonization/VariableDetails.vue

```diff
@@ -11,60 +13,110 @@ const props = defineProps<{
   cohortsWithMapping: { cohort: { id: string }; status: HarmonizationStatus }[];
 }>();
 
-const activeIndex = ref(0);
+const activeTabIndex = ref(0);
 
-const statusPerCohort = computed(
-  () =>
-    calcHarmonizationStatus(
-      [props.variable],
-      props.cohortsWithMapping.map((cwm) => cwm.cohort)
-    )[0]
+const statusPerCohort = computed(() =>
+  calcIndividualVariableHarmonizationStatus(
+    props.variable,
+    props.cohortsWithMapping.map((cwm) => cwm.cohort)
+  )
 );
 
 const activeCohortId = computed(
-  () => props.cohortsWithMapping[activeIndex.value].cohort.id
+  () => props.cohortsWithMapping[activeTabIndex.value].cohort.id
 );
 
-const activeMappingDescription = computed(() => {
-  const activeCohortMapping = props.variable?.mappings?.find(
-    (mapping) => mapping.sourceDataset.resource.id === activeCohortId.value
-  );
-  return activeCohortMapping?.description || "No description";
+const activeMappingDescriptions = computed(() => {
+  if (props.variable.repeats) {
+    const baseMapping = props.variable.mappings?.find(
+      (mapping) => mapping.sourceDataset.resource.id === activeCohortId.value
+    )?.description;
+    const repeatMappings = props.variable.repeats.map((repeat) => {
+      return repeat.mappings?.find(
+        (mapping) => mapping.sourceDataset.resource.id === activeCohortId.value
+      )?.description;
+    });
+    return [baseMapping, ...repeatMappings];
```

Comment:
nice

Reply:
well , these is some discussion if the baseMapping should be in the list  , suggest to fix ( if needed ) in separate issue, this pr should really have been split into multiple as it's really a fix of fixes and features :S